pkg/operator/reporting/templates.go

```diff
@@ -129,10 +130,32 @@ func (ctx *ReportQueryTemplateContext) newQueryTemplate() (*template.Template, e
 // and checks if the returned error is nil. If nil, return an empty string and the error, else return
 // the function call to renderTemplate, passing in the new query template and tmplCtx parameter.
 func RenderQuery(ctx *ReportQueryTemplateContext, tmplCtx TemplateContext) (string, error) {
+	// validate the ctx.ReportQuery.Spec.Query field and create a new template
 	tmpl, err := ctx.newQueryTemplate()
 	if err != nil {
 		return "", err
 	}
+
+	// validate that all required inputs were passed
+	expectedVals := sets.NewString()
+	expectedInputs := sets.NewString()
+
+	for _, queryDef := range ctx.ReportQuery.Spec.Inputs {
+		// check whether the current definition is required
+		if queryDef.Required {
```

Comment:
We may need a set of required, and optional inputs and compare the provided values against each, because as-is, I believe we would get errors for providing inputs that aren't required. Tests can confirm though.